### PR TITLE
Insert bytecode in `EVM` for `asset-metadata-extender` precompile

### PR DESCRIPTION
**Pull request by asiniscalchi**
- Added a constant `STORAGE_VERSION` to represent the current storage version in both `asset-metadata-extender` and `laos-evolution` pallets.
- Added `AddPrecompileBytecode` migration that inserts bytecode

### CODE REVIEW

File: ownership-chain/runtime/src/migration.rs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		}
+
+		let db_weight = <T as frame_system::Config>::DbWeight::get();
+		let mut consumed_weight = Default::default();
+
+		Evm::<T>::create_account(
+			asset_metadata_extender_address,
+			pallet_evm_evolution_collection_factory::REVERT_BYTECODE.into(),
+		);
+		consumed_weight += db_weight.writes(1);
```
Comment (dastanfv) [inline ownership-chain/runtime/src/migration.rs:48]: this is two writes: `AccountCodes` and `AccountMetadata`. And also does two reads

File: ownership-chain/runtime/src/migration.rs [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	Evm::<T>::account_code_metadata(address).size != 0
+}
+
+impl<T> OnRuntimeUpgrade for AddPrecompileBytecode<T>
+where
+	T: pallet_evm::Config,
+{
+	#[cfg(feature = "try-runtime")]
+	fn pre_upgrade() -> Result<Vec<u8>, sp_runtime::TryRuntimeError> {
+		let asset_metadata_extender_address: H160 = H160::from_low_u64_be(1029);
```
Comment (dastanfv) [inline ownership-chain/runtime/src/migration.rs:24]: could probably use `hash` function from the `precompile-utils`, to be more consistent

File: ownership-chain/runtime/src/migration.rs [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	traits::{Get, OnRuntimeUpgrade},
+	weights::Weight,
+};
+use pallet_evm::Pallet as Evm;
+use sp_core::H160;
+
+#[cfg(feature = "try-runtime")]
+use sp_std::vec::Vec;
+
+pub struct AddPrecompileBytecode<T>(sp_std::marker::PhantomData<T>);
```
Comment (dastanfv) [inline ownership-chain/runtime/src/migration.rs:12]: some comment would be nice

Comment (dastanfv) [inline ownership-chain/runtime/src/migration.rs:12]: Also, maybe name it `InjectDamePrecompileBytecode` or something more concrete?

File: ownership-chain/runtime/src/migration.rs (hunk 1 above)
Comment (dastanfv) [reply ownership-chain/runtime/src/migration.rs:48]: see in `EvolutionCollectionFactory::create_collection()`, we manually record the weight for this there as well

**Review by dastanfv — CHANGES_REQUESTED** [review]

File: ownership-chain/runtime/src/migration.rs [hunk 4]
```diff
@@ -0,0 +1,63 @@
+use frame_support::{
```
Comment (dastanfv) [inline ownership-chain/runtime/src/migration.rs:1]: is this duplicate of `./precompiles/migration.rs`?

File: ownership-chain/runtime/src/precompiles/mod.rs [hunk 5]
```diff
@@ -116,7 +118,7 @@ where
 	}
 }
 
-fn hash(a: u64) -> H160 {
+pub fn hash(a: u64) -> H160 {
```
Comment (dastanfv) [inline ownership-chain/runtime/src/precompiles/mod.rs:121]: small nit here:
```suggestion
pub(crate) fn hash(a: u64) -> H160 {
```

**Review by dastanfv — APPROVED** [review]

**Review by magecnion — APPROVED** [review]

**Pull request merged**